### PR TITLE
[AND-123] Add version catalogs and type safe project accessors

### PR DESCRIPTION
**Pull request by pedroagostinhoaptoide**
**What does this PR do?**

   - Adds a versions catalog to declare dependencies and plugins between projects.
   - Moves the convention plugins from buildSrc to a new build-logic module.

**Database changed?**

   No

**Where should the reviewer start?**

- [ ] See by commit.

**How should this be manually tested?**

  Flow on how to test this or QA Tickets related to this use-case: [AND-123](https://aptoide.atlassian.net/browse/AND-123)

**What are the relevant tickets?**

  Tickets related to this pull-request: [AND-123](https://aptoide.atlassian.net/browse/AND-123)

**Code Review Checklist**

- [ ] Documentation on public interfaces
- [ ] Database changed? If yes - Migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] New Kotlin code has unit tests
- [ ] New flows in presenters unit tests
- [ ] Mappers/Validators with any kind of logic unit tests
- [ ] Functional tests pass

[AND-123]: https://aptoide.atlassian.net/browse/AND-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[AND-123]: https://aptoide.atlassian.net/browse/AND-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by G0retZ — COMMENTED** [review]
Can we have TOML file inside the `build-logic`?

File: build-logic/convention/src/main/kotlin/plugins/AndroidModulePlugin.kt [hunk 1]
```diff
@@ -1,12 +1,10 @@
-package plugin
+package plugins
```
Comment (G0retZ) [inline build-logic/convention/src/main/kotlin/plugins/AndroidModulePlugin.kt:1]: Why renaming the package? 🤔

File: build-logic/convention/src/main/kotlin/JavaLibrary.kt [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import org.gradle.api.JavaVersion
+
+object JavaLibrary {
+  val SOURCE_COMPATIBILITY_JAVA_VERSION = JavaVersion.VERSION_17
```
Comment (G0retZ) [inline build-logic/convention/src/main/kotlin/JavaLibrary.kt:4]: these are used in 1 place only.
Lets set those values directly and get rid of this file :-)

**Review by G0retZ — CHANGES_REQUESTED** [review]

File: build-logic/convention/src/main/kotlin/plugins/AndroidModulePlugin.kt (hunk 1 above)
Comment (pedroagostinhoaptoide) [reply build-logic/convention/src/main/kotlin/plugins/AndroidModulePlugin.kt]: Fixed

File: build-logic/convention/src/main/kotlin/JavaLibrary.kt (hunk 2 above)
Comment (pedroagostinhoaptoide) [reply build-logic/convention/src/main/kotlin/JavaLibrary.kt]: Fixed

**Review by G0retZ — APPROVED** [review]

**Pull request merged**